File: packages/shared/src/types/tenant.ts
export type Region = 'RU' | 'GLOBAL';
export type LlmProvider = 'yandex' | 'claude';
export type TenantPlan = 'starter' | 'pro' | 'enterprise';

export interface Tenant {
  id: string;
  slug: string;
  name: string;
  region: Region;
  llmProvider: LlmProvider;
  brandColor: string;
  logoUrl: string | null;
  language: string;
  widgetApiKey: string;
  plan: TenantPlan;
  createdAt: string;
}

export interface TenantUser {
  id: string;
  tenantId: string;
  email: string;
  role: 'admin' | 'viewer';
  createdAt: string;
}

File: apps/api/app/schemas/tenant.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class TenantOut(BaseModel):
    id: UUID
    slug: str
    name: str
    region: Literal["RU", "GLOBAL"]
    llm_provider: Literal["yandex", "claude"]
    brand_color: str
    logo_url: str | None
    language: str
    plan: str
    created_at: datetime
